doc/default.nix

```diff
@@ -16,40 +13,75 @@ let
   # Prefix to remove from option declaration file paths.
   rootPrefix = toString ../. + "/";
 
-  nixosConfiguration = nixosSystem {
-    inherit system;
-    modules = [
-      inputs.home-manager.nixosModules.home-manager
-      inputs.self.nixosModules.stylix
-      ./settings.nix
-    ];
+  # A stub pkgs used while evaluating the stylix modules for the docs
+  noPkgs = {
+    # Needed for type-checking
+    inherit (pkgs) _type;
+
+    # Permit access to (pkgs.formats.foo { }).type
+    formats = lib.mapAttrsRecursive (_: fmt: args: {
+      inherit (fmt args) type;
+    }) pkgs.formats;
   };
 
-  homeConfiguration = homeManagerConfiguration {
-    inherit pkgs;
-    modules = [
-      inputs.self.homeModules.stylix
-      ./settings.nix
-      {
-        home = {
-          homeDirectory = "/home/book";
-          stateVersion = "22.11";
-          username = "book";
+  # A stub config used while evaluating the stylix modules for the docs
+  #
+  # To resolve #98, this should be simplified to `noConfig = {}`.
+  # However, currently we access option-declaring functions via
+  # `config.lib.stylix`.
+  noConfig =
+    let
+      configuration = evalDocs {
```

Comment:
Note, this doesn't _strictly_ need to be done in a separate eval, however I found it cleaner and simpler to write it this way.

With this approach, the lib functions are evaled once, rather than per-platform.

Additionally, recursively inheriting from the "actual" `configuration.config` ended up with messier code, hence why I separated it out.

Once the option-declaring functions are accessed another way, this can be removed and replaced with `specialArgs.config = { }`.

Reply:
If you also wanted to [scrub derivations](https://github.com/NixOS/nixpkgs/blob/d6a5810feb1772e665138ba97ecf1ff033e7feb6/nixos/modules/misc/documentation.nix#L90-L106) from the `options` set, you'd need to refactor `evalDocs` to be somewhat recursive:

```nix
evalDocs =
  module:
  let
    configuration = lib.evalModules {
      modules = # ...
      specialArgs = {
        options = scrubDerivations configuration.options;
        # ...
      };
    };
  in
  configuration;
```

However I'm not convinced this is needed.